src/metabase/api/x_ray.clj

```diff
@@ -32,163 +33,133 @@
+  (when-not (= database database/virtual-id)
+    (api/read-check Database database))
+  (->> query
+       virtual-card
        (fe/extract-features {:max-cost (max-cost max_query_cost
                                                  max_computation_cost)})
        fe/x-ray))
 
-(api/defendpoint GET "/compare/fields/:id1/:id2"
-  "Get comparison x-ray for `Field`s with ID1 and ID2."
-  [id1 id2 max_query_cost max_computation_cost]
+(api/defendpoint GET "/compare/:models/:id1/:id2"
+  "Get comparison x-ray for two models of type `:models` with ids :id1 and id2."
+  [models id1 id2 max_query_cost max_computation_cost]
   {max_query_cost       MaxQueryCost
-   max_computation_cost MaxComputationCost}
+   max_computation_cost MaxComputationCost
+   models               Model}
   (api/check-403 (costs/enable-xrays))
   (->> [id1 id2]
-       (map #(api/read-check Field (Integer/parseInt %)))
+       (map #(api/read-check (->model models) (Long/parseLong %)))
        (apply fe/compare-features
               {:max-cost (max-cost max_query_cost max_computation_cost)})
        fe/x-ray))
 
-(api/defendpoint GET "/compare/tables/:id1/:id2"
-  "Get comparison x-ray for `Table`s with ID1 and ID2."
-  [id1 id2 max_query_cost max_computation_cost]
+(api/defendpoint GET "/compare/:model1/:id1/model2/:id2"
+  "Get comparison x-ray for two models of types `:model1` and `:model2` with
+   ids `:id1` and `:id2`."
+  [model1 model2 id1 id2 max_query_cost max_computation_cost]
   {max_query_cost       MaxQueryCost
-   max_computation_cost MaxComputationCost}
+   max_computation_cost MaxComputationCost
+   model1               Model
+   model2               Model}
   (api/check-403 (costs/enable-xrays))
-  (->> [id1 id2]
-       (map #(api/read-check Table (Integer/parseInt %)))
-       (apply fe/compare-features
-              {:max-cost (max-cost max_query_cost max_computation_cost)})
-       fe/x-ray))
+  (fe/x-ray
+   (fe/compare-features
+    {:max-cost (max-cost max_query_cost max_computation_cost)}
+    (api/read-check (->model model1) (Long/parseLong id1))
+    (api/read-check (->model model2) (Long/parseLong id2)))))
 
-(api/defendpoint GET "/compare/tables/:id1/:id2/field/:field"
-  "Get comparison x-ray for `Field` named `field` from `Table`s with ID1 and
-   ID2."
-  [id1 id2 field max_query_cost max_computation_cost]
+(api/defendpoint GET "/compare/:model/:id1/:id2/field/:field"
+  "Get comparison x-ray for `Field` named `field` from models of type `:model`
+   with ids `:id1` and `:id2`."
+  [model id1 id2 field max_query_cost max_computation_cost]
   {max_query_cost       MaxQueryCost
-   max_computation_cost MaxComputationCost}
+   max_computation_cost MaxComputationCost
+   model                Model}
   (api/check-403 (costs/enable-xrays))
   (let [{:keys [comparison constituents]}
         (->> [id1 id2]
-             (map #(api/read-check Table (Integer/parseInt %)))
+             (map #(api/read-check (->model model) (Long/parseLong %)))
```

Comment:
if you just give them a name that *ends* in ID then the `defendpoint` magic will take care of parsing them for you. You could do something like name them `first-id` `second-id` or something.